tests/functional/warehouse_test/test_warehouses.py

```diff
@@ -90,3 +89,47 @@ def test_snowflake_override_ok(self, project):
             ]
         )
         check_relations_equal(project.adapter, ["OVERRIDE_WAREHOUSE", "EXPECTED_WAREHOUSE"])
+
+
+class TestInvalidConfigWarehouse:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "invalid_warehouse.sql": project_config_models__warehouse_sql,
+        }
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "models": {
+                "test": {"snowflake_warehouse": "DBT_TEST_DOES_NOT_EXIST"},
+            },
+        }
+
+    def test_snowflake_override_invalid(self, project):
+        result = run_dbt(["run", "--models", "invalid_warehouse"], expect_pass=False)
+        assert "Object does not exist, or operation cannot be performed" in result[0].message
+
+
+class TestValidConfigWarehouse:
+    @pytest.fixture(scope="class")
```

Comment:
I think you just need to define this and the other model in outside of all classes once.